### PR TITLE
Support for Hapi v17 (drops support for prior versions, use 1.x)

### PR DESCRIPTION
**Pull request by rajivm**
- bump version to 2.0.0

### CODE REVIEW

**Comment by jamesdixon** [comment]
LGTM!

File: lib/index.js [hunk 1]
```diff
@@ -54,9 +50,11 @@ function onPostAuthFactory(qsOptions) {
         request.payload = qs.parse(request.payload, qsOptions);
     }
 
-    return reply.continue();
+    return h.continue;
   };
 };
 
+register.attributes = { pkg: require('../package.json') };
+exports.name = require('../package.json')['name'];
```
Comment (dafortune) [inline lib/index.js]: Are you sure this works? From hapi release notes https://github.com/hapijs/hapi/issues/3658 (Plugins section, sorry I cannot link it directly) 
```
Replace the exports.register() and the matching exports.register.attributes with exports.plugin = { register, name, version, multiple, dependencies, once, pkg }
```

looks like the expected object is 
```
exports.plugin = { 
  register, 
  pkg: require('../package.json')
}
```

File: tests/features/parse-payload.test.js [hunk 2]
```diff
@@ -4,46 +4,69 @@ const basicServer = require('../support/basic_server');
...
+          if (isMultipart) {
+            
+              let formData = new FormData();
+              for (let key in form) {
+                formData.append(key, form[key]);
+              }
+              requestOptions.headers = formData.getHeaders();
+              formData = await streamToPromise(formData);
+              requestOptions.payload = formData;
+              //console.log(requestOptions.payload);
```
Comment (dafortune) [inline tests/features/parse-payload.test.js]: Please remove :)

**Review by dafortune — COMMENTED** [review]
Thanks a lot for the PR! I've made just minor comments, then it is ready to merge. I need to update the readme too, because v16 is going to be here for a while yet.

File: lib/index.js (hunk 1 above)
Comment (rajivm) [reply lib/index.js]: I will verify again -- but we are using this in our project :) The Hapi docs on the migration are not perfect is what I discovered.

**Comment by rajivm** [comment]
@dafortune Thanks -- will make these changes & push up later today

File: lib/index.js (hunk 1 above)
Comment (rajivm) [reply lib/index.js]: So it looks like the new syntax works as well, so I moved towards that, thanks for noticing :) Also looks like the plugin docs are still out of date.

**Comment by rajivm** [comment]
@dafortune Made updates including README

**Comment by dafortune** [comment]
Thanks again! Published as 2.0.0

**Pull request merged**